pages/cart.tsx

```diff
@@ -35,109 +39,78 @@ export default function Cart() {
     }
   )
 
-  const items = data?.line_items.physical_items ?? []
-
-  const error = null
-  const success = null
-
   return (
-    <div className="grid lg:grid-cols-12">
-      <div className="lg:col-span-8">
-        {isEmpty ? (
-          <div className="flex-1 px-12 py-24 flex flex-col justify-center items-center ">
-            <span className="border border-dashed border-secondary rounded-full flex items-center justify-center w-16 h-16 bg-primary p-12 rounded-lg text-primary">
-              <Bag className="absolute" />
-            </span>
-            <h2 className="pt-6 text-2xl font-bold tracking-wide text-center">
-              Your cart is empty
-            </h2>
-            <p className="text-accents-6 px-10 text-center pt-2">
-              Biscuit oat cake wafer icing ice cream tiramisu pudding cupcake.
-            </p>
-          </div>
-        ) : error ? (
-          <div className="flex-1 px-4 flex flex-col justify-center items-center">
-            <span className="border border-white rounded-full flex items-center justify-center w-16 h-16">
-              <Cross width={24} height={24} />
-            </span>
-            <h2 className="pt-6 text-xl font-light text-center">
-              We couldn’t process the purchase. Please check your card
-              information and try again.
-            </h2>
-          </div>
-        ) : success ? (
-          <div className="flex-1 px-4 flex flex-col justify-center items-center">
-            <span className="border border-white rounded-full flex items-center justify-center w-16 h-16">
-              <Check />
-            </span>
-            <h2 className="pt-6 text-xl font-light text-center">
-              Thank you for your order.
-            </h2>
-          </div>
-        ) : (
-          <div className="px-4 sm:px-6 flex-1">
-            <Text variant="pageHeading">My Cart</Text>
-            <Text variant="sectionHeading">Review your Order</Text>
-            <ul className="py-6 space-y-6 sm:py-0 sm:space-y-0 sm:divide-y sm:divide-accents-2 border-b border-accents-2">
-              {items.map((item) => (
-                <CartItem
-                  key={item.id}
-                  item={item}
-                  currencyCode={data?.currency.code!}
-                />
-              ))}
-            </ul>
-            <div className="my-6">
-              <Text>
+    <div className="px-4 pt-2 sm:px-6 flex-1">
+      <Text variant="pageHeading">My Cart</Text>
+      {error ? (
+        <div className="mt-2">Failed to load</div>
+      ) : isLoading ? (
+        <CartSkeleton />
+      ) : items.length > 0 ? (
+        <div className="grid lg:grid-cols-12">
```

Comment:
What do you think of isolating this part into a new component? You can get advantages of composition for this case

```jsx
{error && <CartError />}
{isLoading && <CartSkeleton />}
{!!items.length && <Cart items={items} />}
```